frontend/Container.js

```diff
@@ -37,14 +43,42 @@ class Container extends React.Component {
     },
     extraTabs: {[key: string]: () => React$Element},
   };
+  state: State;
+
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      isVertical: (window.innerWidth < 500),
+    };
+  }
+
+  componentDidMount() {
+    window.addEventListener('resize', (e) => this.handleResize(e), false);
+    this.setState({
+      isVertical: (window.innerWidth < 500),
+    });
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize');
```

Comment:
I actually prefer this syntax, but have had trouble getting it to work in other people's projects before, a simple fix.